token-analytics/internal/quicknode/.quicknode-streams-filters/bonding_curve_smart_contract.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * SPDX-License-Identifier: ice License 1.0
+ */
+
+//Blocks with receipt dataset
+
+function main(payload) {
+    const {
+        data,
+        metadata,
+    } = payload;
+    txs = {}
+    filtered = data[0].receipts.filter(function(item) {
+        logs = item.logs.filter(function(log) {
+            // Contract addr
+            contains = log.address.toLowerCase() === '{{.ContractAddress}}'.toLowerCase()
+            if (contains) {
+                txs[log.transactionHash] = true
+            }
+            return contains
+        })
+
+        return logs.length > 0
+    })
+    const relevantTransactions = data[0].block.transactions.filter(tx => txs[tx.hash] === true);
+    
+    relevantTransactions.forEach(tx => {
+        tx.blockHash = data[0].block.hash;
+        tx.blockTimestamp = data[0].block.timestamp;
+    });
+    
+    return {
+        'transactions': relevantTransactions,
+        'logs': filtered.map(receipt => receipt.logs).flat(2),
```

Comment:
what if we add the logs inside each transaction element and remove the transactionHash from the log object ? @ice-myles @ice-perseus